climada/util/coordinates.py

```diff
@@ -1491,8 +1505,9 @@ def read_raster(file_name, band=None, src_crs=None, window=None, geometry=None,
         number of lons for transform
     height : float
         number of lats for transform
-    resampling : rasterio.warp.Resampling optional
-        resampling function used for reprojection to dst_crs
+    resampling : int, rasterio.enums.Resampling or str, optional
+        Resampling method to use. String values like `"nearest"` or `"bilinear"` are resolved to
+        attributes of `rasterio.enums.Resampling. Default: "nearest"
```

Comment:
I copy my answer from https://github.com/CLIMADA-project/climada_python/pull/339#discussion_r779407184: 

The enum `rasterio.enums.Resampling` actually has integer values (https://rasterio.readthedocs.io/en/latest/api/rasterio.enums.html#rasterio.enums.Resampling), so the type of function parameters is `int, rasterio.enums.Resampling`, see e.g. the docstring of `rasterio.warp.reproject` (https://rasterio.readthedocs.io/en/latest/api/rasterio.warp.html#rasterio.warp.reproject).

Reply:
That is not my question :D This is not explained in the docstring clearly enough imho :D

Reply:
It is said what happens with string values, or with the resampling method. But what happens with integers?

Reply:
As I said, if you put `resampling=rasterio.enums.Resampling.bilinear` than this means you are actually providing an int, because `rasterio.enums.Resampling.bilinear` literally is the integer value `1` (see https://rasterio.readthedocs.io/en/latest/api/rasterio.enums.html#rasterio.enums.Resampling). So, the kwarg is actually accepting integer values. As I said, this is the same docstring format as used by rasterio (https://rasterio.readthedocs.io/en/latest/api/rasterio.warp.html#rasterio.warp.reproject).

Reply:
Thanks for the explanation. I think it is a bit cryptic how one should know that `rasterio.enums.Resampling`  is an integer, but it is consistent.

Reply:
My intention was to use a docstring that is consistent with the docstring used by rasterio. However, I see that this confuses users, so I changed it to `int or str` and added an explanation:
https://github.com/CLIMADA-project/climada_python/blob/b74a06446e153b36dea8ab5f948002858160b168/climada/util/coordinates.py#L1506-L1509